lib/whitehall/asset_manager_storage.rb

```diff
@@ -50,13 +50,19 @@ def initialize(asset_path, model = nil, version_name = nil)
     end
 
     def delete
-      AssetManagerDeleteAssetWorker.perform_async(@legacy_url_path, nil)
+      if use_non_legacy
```

Comment:
I prefer the previous `should_have_non_legacy_urls?` naming here. Either way, do use the ? for the method name.

Also below 👇🏻 maybe we can refactor a bit with a return unless.

Reply:
added '?'